### PR TITLE
refactor(filemanager): API style

### LINKED ISSUE #458: filemanager: update API style
The filemanager API should be updated to match the style of other APIs, such as:

* `camalCase` parameters and responses.
* `swagger_ui` -> `swagger-ui`.
* Responses/pagination with the following structure:
```json
{
    "links": {
        "next": "<...>",
        "previous": "<...>",
    },
    "pagination": {
        "count": "<...>",
        "page": "<...>",
        "rowsPerPage": "<...>",
    },
    "results": [ "<...>" ],
}
```
* `-` preferred over `_` in route paths? Or best to use a single word if possible?

### PR DESCRIPTION
**Pull request by mmalenic**
Closes #458

### Changes
* All responses/requests are `camelCase`.
* Routes are shortened to not use `-` and `_`.
    * Swagger ui uses `swagger-ui` instead of `swagger_ui`.
* Pagination uses links and contains count.

### CODE REVIEW

File: lib/workload/stateless/stacks/filemanager/docs/API_GUIDE.md [hunk 1]
```diff
@@ -91,21 +99,21 @@ on a key prefix:
...
+"https://file.dev.umccr.org/api/v1/s3" | jq
 ```
 
-Case-insensitive wildcard matching, which gets converted to a postgres `ilike` statement, is supported by using `case_sensitive`:
+Case-insensitive wildcard matching, which gets converted to a postgres `ilike` statement, is supported by using `caseSensitive`:
 
 ```sh
 curl --get -H "Authorization: Bearer $TOKEN" --data-urlencode "key=temp\_data%" \
-"https://file.dev.umccr.org/api/v1/s3_objects?case_sensitive=false" | jq
+"https://file.dev.umccr.org/api/v1/s3?case_sensitive=false" | jq
```
Comment (victorskl) [inline lib/workload/stateless/stacks/filemanager/docs/API_GUIDE.md:109]:
```
"https://file.dev.umccr.org/api/v1/s3?caseSensitive=false" | jq
```

File: lib/workload/stateless/stacks/filemanager/filemanager-api-lambda/README.md [hunk 2]
```diff
@@ -8,4 +8,7 @@ the relevant token:
 curl -H "Authorization: Bearer $orcabus_api_token" https://file.<stage>.umccr.org/api/v1/s3_objects/count
```
Comment (victorskl) [inline lib/workload/stateless/stacks/filemanager/filemanager-api-lambda/README.md:8]:
```
curl -H "Authorization: Bearer $orcabus_api_token" https://file.<stage>.umccr.org/api/v1/s3/count
```

**Review by victorskl — APPROVED** [review]
LGTM. Just minor comments.

File: lib/workload/stateless/stacks/filemanager/filemanager-api-lambda/README.md (hunk 2 above)
Comment (mmalenic) [reply lib/workload/stateless/stacks/filemanager/filemanager-api-lambda/README.md]: Thanks! Good catch.

**Pull request merged**